pkg/resources/certificates/certificate.go

```diff
@@ -0,0 +1,60 @@
+package certificates
+
+import (
+	v1alpha1 "github.com/interconnectedcloud/qdrouterd-operator/pkg/apis/interconnectedcloud/v1alpha1"
+	cmv1alpha1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Create NewIssuerForCR method to create issuer
+func NewIssuerForCR(m *v1alpha1.Qdrouterd, config string) *cmv1alpha1.Issuer {
+	issuerConfig := cmv1alpha1.IssuerConfig{}
+	issuerExt := "-ss"
+	if config == "ca" {
+		issuerExt = "-ca"
+		issuerConfig.CA = &cmv1alpha1.CAIssuer{
+			SecretName: m.Spec.Issuer,
+		}
+	} else {
+		issuerConfig.SelfSigned = &cmv1alpha1.SelfSignedIssuer{}
+	}
+	issuer := &cmv1alpha1.Issuer{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "certmanager.k8s.io/v1alpha1",
+			Kind:       "Issuer",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      m.Name + issuerExt,
+			Namespace: m.Namespace,
+		},
+		Spec: cmv1alpha1.IssuerSpec{
+			IssuerConfig: issuerConfig,
+		},
+	}
+
+	return issuer
+}
+
+// Create NewCertificateForCR method to create new certificate
+func NewCertificateForCR(m *v1alpha1.Qdrouterd, issuerName string) *cmv1alpha1.Certificate {
+	cert := &cmv1alpha1.Certificate{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "certmanager.k8s.io/v1alpha1",
+			Kind:       "Certificate",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      m.Name + "-" + issuerName,
+			Namespace: m.Namespace,
+		},
+		Spec: cmv1alpha1.CertificateSpec{
+			SecretName: m.Name + "-" + issuerName + "-tls",
+			CommonName: m.Name + "." + m.Namespace + ".svc.cluster.local",
```

Comment:
Ideally we will also populate the dnsNames field with the service name on its own and, where applicable, the hostname from any associated route. That could be done in a subsequent update, but it is one of the reasons for using the cert manager instead of just requesting a server cert from openshift.